core/designsystem/src/main/java/com/acon/core/designsystem/component/textfield/AconSearchBar.kt

```diff
@@ -0,0 +1,171 @@
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.onFocusChanged
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalFocusManager
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.acon.core.designsystem.R
+import com.acon.core.designsystem.theme.AconColors
+import com.acon.core.designsystem.theme.AconTheme
+
+@Composable
+fun AconSearchBar(
+    status: SearchBarStatus,
+    modifier: Modifier = Modifier,
+    text: String = "text",
+    placeholder: String = "",
+    onTextChanged: (String) -> Unit = {},
+    onFocusChanged: (Boolean) -> Unit = {},
+    clearText: () -> Unit = {},
+    keyboardOptions: KeyboardOptions = KeyboardOptions.Default,
+    keyboardActions: KeyboardActions = KeyboardActions.Default,
+    leadingIcon: @Composable () -> Unit = {
+        Row {
+            Icon(
+                painter = painterResource(id = R.drawable.ic_search_24),
+                contentDescription = "Search",
+                tint = AconColors.White
+            )
+            Spacer(modifier = Modifier.width(6.dp))
+        }
+    },
+    tailingIcon: @Composable () -> Unit = {
+        if (status == SearchBarStatus.Active) {
+            Icon(
+                painter = painterResource(id = R.drawable.ic_dissmiss_circle_gray),
+                contentDescription = "Clear",
+                tint = Color.Unspecified,
+                modifier = Modifier
+                    .size(20.dp)
+                    .clickable { clearText() }
+            )
+        }
+    },
+) {
+    val focusRequester = remember { FocusRequester() }
+    val focusManager = LocalFocusManager.current
+
+    val backgroundColor = when (status) {
+        SearchBarStatus.Inactive -> AconColors.Gray8
+        SearchBarStatus.Focused -> AconColors.Gray8
+        SearchBarStatus.Active -> AconColors.Gray8
+    }
+
+    val borderColor = when (status) {
+        SearchBarStatus.Inactive -> AconColors.Gray6
+        SearchBarStatus.Focused -> AconColors.Gray5
+        SearchBarStatus.Active -> AconColors.Gray5
+    }
+
+    val textColor = when (status) {
+        SearchBarStatus.Inactive -> AconColors.Gray5
+        SearchBarStatus.Focused -> AconColors.Gray5
+        SearchBarStatus.Active -> AconColors.White
+    }
+
+    Box(
+        modifier = modifier
+            .fillMaxWidth()
+            .height(40.dp)
+            .background(backgroundColor, shape = RoundedCornerShape(8.dp))
+            .border(1.dp, borderColor, shape = RoundedCornerShape(8.dp))
```

Comment:
named argument 처리는 모든 파라미터에 해주셔야 헷갈릴 여지가 줄어들 거 같아요!
```suggestion
            .border(width = 1.dp, color = borderColor, shape = RoundedCornerShape(8.dp))
```